test/releasers/node.ts

```diff
@@ -0,0 +1,201 @@
+    // the delta since the last release.
+    .get('/repos/googleapis/node-test-repo/tags?per_page=100')
+    .reply(200, [
+      {
+        name: 'v0.123.4',
+        commit: {
+          sha: 'da6e52d956c1e35d19e75e0f2fdba439739ba364',
+        },
+      },
+    ])
+    .post('/graphql')
+    .reply(200, {
+      data: graphql,
+    })
+    // getting the latest tag
+    .get('/repos/googleapis/node-test-repo/git/refs?per_page=100')
+    .reply(200, [{ref: 'refs/tags/v0.123.4'}])
+    // creating a new branch
+    .post('/repos/googleapis/node-test-repo/git/refs')
+    .reply(200)
+    // check for CHANGELOG
+    .get(
+      '/repos/googleapis/node-test-repo/contents/CHANGELOG.md?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(404)
+    .put(
+      '/repos/googleapis/node-test-repo/contents/CHANGELOG.md',
+      (req: {[key: string]: string}) => {
+        snapshot(`CHANGELOG-node-message-${snapName}`, req.message);
+        snapshot(
+          `CHANGELOG-node-${snapName}`,
+          Buffer.from(req.content, 'base64')
+            .toString('utf8')
+            .replace(/\([0-9]{4}-[0-9]{2}-[0-9]{2}\)/g, '')
+        );
+        return true;
+      }
+    )
+    .reply(201)
+    // update package.json
+    .get(
+      '/repos/googleapis/node-test-repo/contents/package.json?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(200, {
+      content: Buffer.from(packageContent, 'utf8').toString('base64'),
+      sha: 'abc123',
+    })
+    .put(
+      '/repos/googleapis/node-test-repo/contents/package.json',
+      (req: {[key: string]: string}) => {
+        snapshot(`package-json-node-message-${snapName}`, req.message);
+        snapshot(
+          `package-json-node-${snapName}`,
+          Buffer.from(req.content, 'base64').toString('utf8')
+        );
+        return true;
+      }
+    )
+    .reply(200)
+    .get(
+      '/repos/googleapis/node-test-repo/contents/samples/package.json?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(404)
+    .get(
+      '/repos/googleapis/node-test-repo/contents/samples/package-lock.json?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(404)
+    // create release
+    .post(
+      '/repos/googleapis/node-test-repo/pulls',
+      (req: {[key: string]: string}) => {
+        const body = req.body.replace(/\([0-9]{4}-[0-9]{2}-[0-9]{2}\)/g, '');
+        snapshot(`PR body-node-${snapName}`, body);
+        return true;
+      }
+    )
+    .reply(200, {number: 1})
+    .post(
+      '/repos/googleapis/node-test-repo/issues/1/labels',
+      (req: {[key: string]: string}) => {
+        snapshot(`labels-node-${snapName}`, req);
+        return true;
+      }
+    )
+    .reply(200, {})
+    // this step tries to close any existing PRs; just return an empty list.
+    .get('/repos/googleapis/node-test-repo/pulls?state=open&per_page=100')
+    .reply(200, []);
+
+  return req;
+}
+
+describe('Node', () => {
+  describe('run', () => {
+    before(requireNode10);
```

Comment:
I don't think we should be testing on Node 8 any longer, so you might be able to drop these gates.

Reply:
Done